realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -599,7 +599,7 @@ public void migration_backlinkedSourceFieldDoesntExist() throws ClassNotFoundExc
                 .build();
         OsObjectSchemaInfo sourceSchemaInfo = new OsObjectSchemaInfo.Builder("BacklinksSource")
                 .addPersistedProperty("name", RealmFieldType.STRING, !Property.PRIMARY_KEY, !Property.INDEXED, !Property.REQUIRED)
-                .addPersistedLinkProperty("child", RealmFieldType.OBJECT, "BacklinksTarget")
+                .addPersistedLinkProperty("child", RealmFieldType.OBJECT, "BacklinksTarget", !Property.REQUIRED)
```

Comment:
@zaki50 changes in this file doesn't seem to be needed?

Reply:
@beeender I'm removing that method in the next part.
https://github.com/realm/realm-java/pull/5210/files#diff-be39bd214889f0b96ba915e9ec8917d0L74
 Having both `addPersistedLinkProperty(String, RealmFieldType, String)` and  `addPersistedLinkProperty(String, RealmFieldType, String, boolean)` is confusing, isn't it?

Reply:
actually for primitive list, `addPersistedProperty()` is more preferred than `addPersistedLinkProperty()` since primitive list is not a link type.

Reply:
Sounds reasonable.

Reply:
@beeender Updated.